tokio/src/runtime/dump.rs

```diff
@@ -30,14 +30,158 @@ pub struct Task {
     trace: Trace,
 }
 
+/// A backtrace symbol. This is similar to [backtrace::BacktraceSymbol],
+/// but is a separate struct to avoid public dependency issues.
+///
+/// This struct is guaranteed to be pure data and operations involving
+/// it will not call platform functions that take an unpredictable amount
+/// of time to finish.
+#[derive(Clone, Debug)]
+pub struct BacktraceSymbol {
+    name: Option<Vec<u8>>,
+    name_demangled: Option<String>,
+    addr: Option<*mut std::ffi::c_void>,
+    filename: Option<std::path::PathBuf>,
+    lineno: Option<u32>,
+    colno: Option<u32>,
+}
+
+impl BacktraceSymbol {
+    pub(crate) fn from_backtrace_symbol(sym: &backtrace::BacktraceSymbol) -> Self {
+        let name = sym.name();
+        Self {
+            name: name.as_ref().map(|name| name.as_bytes().into()),
+            name_demangled: name.map(|name| format!("{}", name)),
+            addr: sym.addr(),
+            filename: sym.filename().map(From::from),
+            lineno: sym.lineno(),
+            colno: sym.colno(),
+        }
+    }
+
+    /// Return the raw name of the symbol.
+    pub fn name_raw(&self) -> Option<&[u8]> {
+        self.name.as_deref()
+    }
+
+    /// Return the demangled name of the symbol.
+    pub fn name_demangled(&self) -> Option<&str> {
+        self.name_demangled.as_deref()
+    }
```

Comment:
Do we need both? I think users can construct demangled one by using `name_raw`.

Reply:
I just had it since `backtrace` had that, and for some reason I thought the API was somewhat private. Apparently `rustc_demangle` is a perfectly fine public API.

Reply:
Ah, Never mind ... I missed you were using the [`Display` impl of `SymbolName`](https://github.com/rust-lang/backtrace-rs/blob/26ab4afa46e889f2f4c81e5f5022f61569f02d16/src/symbolize/mod.rs#L368-L383) to get a demangle name in 991434ab250bc6d83a0c9f1e4ad297cbb06d5f24.  (I left comment above just because it looked like similar two fields were being added for the same already demangled data.)

To be clear, I think it's fine (or even better) to provide a `name` API that returns the demangled name, as the backtrace crate already [does](https://docs.rs/backtrace/latest/backtrace/struct.BacktraceSymbol.html#method.name).

Reply:
So would you rather have the PR this way or with the `name` and `raw_name` functions?

Reply:
I confirmed that the backtrace crates provide both apis (mangle, demangle), so having `name_raw` and `name_demangled` seems fine to me.